### PR TITLE
Add non-preview notifications endpoints

### LINKED ISSUE #92: Non-Preview Notifications Endpoints
See the endpoints doc for more details about what is open. The subsections for this correspond to the folders under `src`:
- [ ] GET
- [ ] POST
- [ ] PATCH
- [ ] PUT
- [ ] DELETE

Sub-Issue of #28

### PR DESCRIPTION
**Pull request by ktomsic**
Fixes #92 

To avoid conflicts in src/client.rs, this is based on top of my search commits. If you'd prefer me to rebase it onto master instead, that's easy enough.

Also, since there weren't too many endpoints in total, I put everything into one commit. If you think it'd be easier to review by request type (GET - 3 endpoints; PUT - 2 endpoints; DELETE - 1 endpoint; PATCH - 1 endpoint), I can split it up into four separate commits.

### CODE REVIEW

File: src/client.rs [hunk 1]
```diff
@@ -259,6 +261,12 @@ impl <'g> GetQueryBuilder<'g> {
     /// Query the gists endpoint
     func_client!(gists, gists::get::Gists<'g>);
 
+    /// Query the search endpoint
+    func_client!(search, search::get::Search<'g>);
+
+    /// Query the search endpoint
```
Comment (akshatgoel) [inline src/client.rs]: nit : Notifications endpoint.

**Comment by mgattozzi** [comment]
Hey @ktomsic thanks for your patience on waiting for this! Would you mind doing it based off of the master branch? I'm hoping to get to your more involved PR soon but I've been busier than I like lately. If it's small enough then it's fine if they're all combined in one!

**Comment by ktomsic** [comment]
Sure.

**Comment by mgattozzi** [comment]
@ktomsic just pinging you on this one that this will need to be rebased to fix

**Comment by ktomsic** [comment]
Done.

**Comment by Dylan-DPC** [comment]
I have added tests for this and confirmed it works fine. The issue with tests is, you need to have some notifications on github for it to run completely. Else it will silently pass and execute only `GET /notifications` endpoint.

**Pull request merged**